incubator/virtualcluster/pkg/syncer/resources/namespace/checker.go

```diff
@@ -76,98 +79,90 @@ func (c *controller) shouldBeGabageCollected(ns *v1.Namespace) bool {
 	return false
 }
 
-// PatrollerDo checks to see if namespaces in super master informer cache and tenant master
-// keep consistency.
 func (c *controller) PatrollerDo() {
 	clusterNames := c.MultiClusterController.GetClusterNames()
-	if len(clusterNames) != 0 {
-		wg := sync.WaitGroup{}
-		for _, clusterName := range clusterNames {
-			wg.Add(1)
-			go func(clusterName string) {
-				defer wg.Done()
-				c.checkNamespacesOfTenantCluster(clusterName)
-			}(clusterName)
-		}
-		wg.Wait()
-	} else {
-		klog.Infof("tenant masters has no clusters, still check pNamespace for gc purpose.")
```

Comment:
This is a behavior change, we should continue to check pNamespace for gc purpose when no cluster exists.

Reply:
you are right